include/mapbox/geometry/wagyu/process_horizontal.hpp

```diff
@@ -180,10 +180,10 @@ active_bound_list_itr<T> process_horizontal_right_to_left(T scanline_y,
         // OK, so far we're still in range of the horizontal Edge  but make sure
         // we're at the last of consec. horizontals when matching with eMaxPair
         if (is_maxima_edge && bnd == bound_max_pair) {
-            if ((*horz_bound)->ring) {
+            if ((*horz_bound)->ring && (*bound_max_pair)->ring) {
```

Comment:
did we still want to keep this additional `&& (*bound_max_pair)->ring` check? if so, we should also add it at L78

Reply:
I think we should keep it still as it could prevent something really nasty from happening in the `add_local_maximum_point` in case one ring didn't exist.